build.zig

```diff
@@ -596,3 +616,242 @@ const autoconf = .{
+    .DEF_PGPORT = 5432,
+    .DEF_PGPORT_STR = "5432",
+    .DLSUFFIX = ".so",
+    .ENABLE_THREAD_SAFETY = 1,
+    .HAVE_APPEND_HISTORY = 1,
+    .HAVE_ATOMICS = 1,
+    .HAVE_BACKTRACE_SYMBOLS = 1,
+    .HAVE_COMPUTED_GOTO = 1,
+    .HAVE_DECL_FDATASYNC = 1,
+    .HAVE_DECL_F_FULLFSYNC = 0,
+    .HAVE_DECL_POSIX_FADVISE = 1,
+    .HAVE_DECL_PREADV = 1,
+    .HAVE_DECL_PWRITEV = 1,
+    .HAVE_DECL_STRNLEN = 1,
+    .HAVE_EXECINFO_H = 1,
+    .HAVE_EXPLICIT_BZERO = null,
+    .HAVE_FSEEKO = 1,
+    .HAVE_GCC__ATOMIC_INT32_CAS = 1,
+    .HAVE_GCC__ATOMIC_INT64_CAS = 1,
+    .HAVE_GCC__SYNC_CHAR_TAS = 1,
+    .HAVE_GCC__SYNC_INT32_CAS = 1,
+    .HAVE_GCC__SYNC_INT32_TAS = 1,
+    .HAVE_GCC__SYNC_INT64_CAS = 1,
+    .HAVE_GETIFADDRS = 1,
+    .HAVE_GETOPT = 1,
+    .HAVE_GETOPT_H = 1,
+    .HAVE_GETOPT_LONG = 1,
+    .HAVE_HISTORY_TRUNCATE_FILE = 1,
+    .HAVE_IFADDRS_H = 1,
+    .HAVE_INET_ATON = 1,
+    .HAVE_INET_PTON = 1,
+    .HAVE_INTTYPES_H = 1,
+    .HAVE_INT_OPTERR = 1,
+    .HAVE_INT_TIMEZONE = 1,
+    .HAVE_LANGINFO_H = 1,
+    .HAVE_LIBM = 1,
+    .HAVE_LIBREADLINE = 1,
+    .HAVE_LOCALE_T = 1,
+    .HAVE_LONG_INT_64 = 1,
+    .HAVE_MEMORY_H = 1,
+    .HAVE_MKDTEMP = 1,
+    .HAVE_POSIX_FADVISE = 1,
+    .HAVE_POSIX_FALLOCATE = 1,
+    .HAVE_PPOLL = 1,
+    .HAVE_PTHREAD = 1,
+    .HAVE_PTHREAD_BARRIER_WAIT = 1,
+    .HAVE_PTHREAD_PRIO_INHERIT = 1,
+    .HAVE_READLINE_HISTORY_H = 1,
+    .HAVE_READLINE_READLINE_H = 1,
+    .HAVE_RL_COMPLETION_MATCHES = 1,
+    .HAVE_RL_COMPLETION_SUPPRESS_QUOTE = 1,
+    .HAVE_RL_FILENAME_COMPLETION_FUNCTION = 1,
+    .HAVE_RL_FILENAME_QUOTE_CHARACTERS = 1,
+    .HAVE_RL_FILENAME_QUOTING_FUNCTION = 1,
+    .HAVE_RL_RESET_SCREEN_SIZE = 1,
+    .HAVE_RL_VARIABLE_BIND = 1,
+    .HAVE_SOCKLEN_T = 1,
+    .HAVE_SPINLOCKS = 1,
+    .HAVE_STDBOOL_H = 1,
+    .HAVE_STDINT_H = 1,
+    .HAVE_STDLIB_H = 1,
+    .HAVE_STRCHRNUL = null,
+    .HAVE_STRERROR_R = null,
+    .HAVE_STRINGS_H = 0,
+    .HAVE_STRING_H = 1,
+    .HAVE_STRNLEN = 1,
+    .HAVE_STRSIGNAL = 1,
+    .HAVE_STRUCT_OPTION = 1,
+    .HAVE_STRUCT_TM_TM_ZONE = 1,
+    .HAVE_SYNCFS = 1,
+    .HAVE_SYNC_FILE_RANGE = null,
+    .HAVE_SYSLOG = 1,
+    .HAVE_SYS_EPOLL_H = 1,
+    .HAVE_SYS_PERSONALITY_H = 1,
+    .HAVE_SYS_PRCTL_H = 1,
+    .HAVE_SYS_SIGNALFD_H = 1,
+    .HAVE_SYS_STAT_H = 1,
+    .HAVE_SYS_TYPES_H = 1,
+    .HAVE_TERMIOS_H = 1,
+    .HAVE_TYPEOF = 1,
+    .HAVE_UNISTD_H = 1,
+    .HAVE_USELOCALE = 1,
+    .HAVE_VISIBILITY_ATTRIBUTE = 1,
+    .HAVE_X86_64_POPCNTQ = 1,
+    .HAVE__BOOL = 1,
+    .HAVE__BUILTIN_BSWAP16 = 1,
+    .HAVE__BUILTIN_BSWAP32 = 1,
+    .HAVE__BUILTIN_BSWAP64 = 1,
+    .HAVE__BUILTIN_CLZ = 1,
+    .HAVE__BUILTIN_CONSTANT_P = 1,
+    .HAVE__BUILTIN_CTZ = 1,
+    .HAVE__BUILTIN_FRAME_ADDRESS = 1,
+    .HAVE__BUILTIN_OP_OVERFLOW = 1,
+    .HAVE__BUILTIN_POPCOUNT = 1,
+    .HAVE__BUILTIN_TYPES_COMPATIBLE_P = 1,
+    .HAVE__BUILTIN_UNREACHABLE = 1,
+    .HAVE__GET_CPUID = 1,
```

Comment:
Ok this is the only one preventing me to build on apple silicon, I guess we should set this conditionally.

We can merge and I'll open another PR